CRM/Core/Smarty.php

```diff
@@ -340,4 +342,14 @@ private function getLocale() {
     return 'en_US';
   }
 
+  /**
+   * Get the compile_check value.
+   *
+   * @return bool
+   */
+  private function isCheckSmartyIsCompiled() {
+    // check for define in civicrm.settings.php as FALSE, otherwise returns TRUE
+    return CRM_Utils_Constant::value('CIVICRM_TEMPLATE_COMPILE_CHECK', TRUE);
```

Comment:
This is a good idea!

*Informational aside*: if you want to use the "Settings" layer instead of a constant, it should work to call `Civi::settings()->get()`, e.g.

```php
  private function isCheckSmartyIsCompiled() {
    return Civi::settings()->get('smarty_compile_check');
  }
```

or

```php
  private function isCheckSmartyIsCompiled() {
    $check = Civi::settings()->get('smarty_compile_check');
    if ($check === 'auto') {
      return TRUE; /* Future: maybe do something more complicated, like checking if we're in production */
    } else {
      return (bool) $check;
    }
  }
```

The one obscure trick there... the setting would need to be whitelisted in `Civi\Core\SettingsManager::getSystemDefaults()`. (The code here isn't *strictly* boot-critical, but just about every extension in the world calls `CRM_Core_Smarty` from `h_c_config`, so it's *de-facto* critical.)

Of course, it's a subjective call whether to use a constant or a setting. I mention all this because most folks probably assume that a constant is the *only* workable thing in this scenario. (I initially assumed that...)  But it's better to know about this trick so that you can choose based on the other merits of constants-vs-settings.

Reply:
My assumption is that we use constants for things where we think some combination of the following is true

1) it's something for sysadmins to change
2) it is likely to be a transitional setting or we are experimenting with a view to maybe making it the default - this is also the case with the support for multiple locks at mysql level